packages/components/src/date-time/time.js

```diff
@@ -120,46 +180,129 @@ class TimePicker extends Component {
-						isToggled={ am === 'AM' }
-						onClick={ this.updateAmPm( 'AM' ) }
-					>
-						{ __( 'AM' ) }
-					</Button>
-					<Button
-						isDefault
-						className="components-time-picker__pm-button"
-						isToggled={ am === 'PM' }
-						onClick={ this.updateAmPm( 'PM' ) }
-					>
-						{ __( 'PM' ) }
-					</Button>
-				</div> }
-			</div>
+			<fieldset
+				className={
+					is12Hour ? 'components-datetime__time components-datetime__time--12-hour' : 'components-datetime__time components-datetime__time--24-hour'
+				}
+			>
+				<legend>{ __( 'Publish at' ) }</legend>
+				<fieldset className="invisible">
+					<legend className="invisible">{ __( 'Date' ) }</legend>
+					<div className="components-datetime__time__form">
+						<div className="components-datetime__time__form__container components-datetime__time__form__container--month">
+							<select
+								aria-label={ __( 'Month' ) }
+								name="month"
+								value={ month }
+								onChange={ this.onChangeMonth }
+								onBlur={ this.updateMonth }
+							>
+								<option value="01">{ __( 'January' ) }</option>
+								<option value="02">{ __( 'February' ) }</option>
+								<option value="03">{ __( 'March' ) }</option>
+								<option value="04">{ __( 'April' ) }</option>
+								<option value="05">{ __( 'May' ) }</option>
+								<option value="06">{ __( 'June' ) }</option>
+								<option value="07">{ __( 'July' ) }</option>
+								<option value="08">{ __( 'August' ) }</option>
+								<option value="09">{ __( 'September' ) }</option>
+								<option value="10">{ __( 'October' ) }</option>
+								<option value="11">{ __( 'November' ) }</option>
+								<option value="12">{ __( 'December' ) }</option>
+							</select>
+						</div>
+						<div key="day-field" className="components-datetime__time__form__container components-datetime__time__form__container--day">
+							<input
+								aria-label={ __( 'Day' ) }
+								type="number"
+								name="day"
+								value={ day }
+								step={ 1 }
+								min={ 1 }
+								onChange={ this.onChangeDay }
+								onBlur={ this.updateDay }
+							/>
+						</div>
+						<div key="year-field" className="components-datetime__time__form__container components-datetime__time__form__container--year">
+							<input
+								aria-label={ __( 'Year' ) }
+								type="number"
+								name="year"
+								step={ 1 }
+								value={ year }
+								onChange={ this.onChangeYear }
+								onBlur={ this.updateYear }
+							/>
+						</div>
+					</div>
+				</fieldset>
+
+				<fieldset>
+					<legend className="invisible">{ __( 'Time' ) }</legend>
+					<div className="components-datetime__time__form">
+						<div className="components-datetime__time__form__container components-datetime__time__form__container--time">
+							<input
+								aria-label={ __( 'Hours' ) }
+								type="number"
+								name="hours"
+								step={ 1 }
+								min={ this.minHours }
```

Comment:
We're passing a function? Is this meant to be assigning the result of the function `this.minHours()` ? The naming is misleading as well, since it seems like it would be a number value assigned to the instance, when in-fact it's a getter function. I think a name like `getMinHours` would better reflect this.